File: Ahorn/entities/bowlPuffer.jl
module VortexHelperBowlPuffer
using ..Ahorn, Maple

@mapdef Entity "VortexHelper/BowlPuffer" BowlPuffer(x::Integer, y::Integer, noRespawn::Bool = false)

const placements = Ahorn.PlacementDict(
    "Pufferfish Bowl (Vortex Helper)" => Ahorn.EntityPlacement(
        BowlPuffer,
		"point"
    ),
	"Pufferfish Bowl (No Respawn) (Vortex Helper)" => Ahorn.EntityPlacement(
        BowlPuffer,
		"point",
		Dict{String, Any}(
			"noRespawn" => true
		)
    )
)

pufferBowlSprite = "objects/VortexHelper/pufferBowl/idle00"

function Ahorn.selection(entity::BowlPuffer)
    x, y = Ahorn.position(entity)
    return Ahorn.Rectangle(x - 11, y - 11, 21, 19)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::BowlPuffer, room::Maple.Room)
    Ahorn.drawSprite(ctx, pufferBowlSprite, 0, -3)
end

end

File: Loenn/entities/bowl_puffer.lua
local utils = require "utils"

local bowlPuffer = {}

bowlPuffer.name = "VortexHelper/BowlPuffer"
bowlPuffer.depth = 100

bowlPuffer.placements = {
    {
        name = "normal",
        data = {
            noRespawn = false,
            explodeTimer = 1.0,
        },
    },
    {
        name = "no_respawn",
        data = {
            noRespawn = true,
            explodeTimer = 1.0,
        },
    }
}

bowlPuffer.texture = "objects/VortexHelper/pufferBowl/idle00"
bowlPuffer.offset = {32, 35}

function bowlPuffer.selection(room, entity)
    return utils.rectangle((entity.x or 0) - 11, (entity.y or 0) - 11, 21, 19)
end

return bowlPuffer
